stripe/src/main/java/com/stripe/android/StripeKtx.kt

```diff
@@ -570,11 +571,122 @@ suspend fun Stripe.confirmPaymentIntent(
  *
  * @return the result if the API result and JSON parsing are successful; otherwise, throw an exception.
  */
-private inline fun <reified APIObject : StripeModel> runApiRequest(
+private inline fun <reified APIObject : StripeModel> runApiRequestCaptureEmptyResult(
     block: () -> APIObject?
 ): APIObject =
     runCatching {
         requireNotNull(block()) {
             "Failed to parse ${APIObject::class.java.simpleName}."
         }
     }.getOrElse { throw StripeException.create(it) }
+
+/**
+ * Get the [PaymentIntentResult] from [Intent] returned via
+ * Activity#onActivityResult(int, int, Intent)}} for PaymentIntent automatic confirmation
+ * (see [confirmPayment]) or manual confirmation (see [handleNextActionForPayment]})
+ *
+ * @param requestCode [Int] code passed from Activity#onActivityResult
+ * @param data [Intent] intent from Activity#onActivityResult
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class,
+)
+suspend fun Stripe.getPaymentIntentResult(
+    requestCode: Int,
+    data: Intent?,
+): PaymentIntentResult {
+    return runApiRequestCaptureIllegalArgumentException(
+        isForPaymentIntentResult(
+            requestCode,
+            data
+        )
+    ) { paymentController.getPaymentIntentResult(data!!) }
```

Comment:
Would it make sense to make `data` non-nullable and avoid forcing a value here? The nullability will be checked in `isForpaymentIntentResult` so it is safe to force the value when we get to this point. It would also simplify the code in the `isFor*IntentResult` checks and fit in nicely with the `PaymentController` interface which only accepts non-nullable `Intent`.